src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ConflictResolution.targets

```diff
@@ -89,4 +90,50 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   </Target>
 
+  <!--
+    _FilterDuplicateCopyLocalPublishAssets
+    Remove CopyLocalPublishAssets items that could overwrite resolved files to publish.
+    -->
+  <Target Name="_FilterDuplicateCopyLocalPublishAssets">
+    <ItemGroup>
+      <_ResolvedCopyLocalPublishAssetsWithMetadata Include="@(_ResolvedCopyLocalPublishAssets->Distinct())">
+        <RelativePath>%(_ResolvedCopyLocalPublishAssets.DestinationSubDirectory)%(Filename)%(Extension)</RelativePath>
+      </_ResolvedCopyLocalPublishAssetsWithMetadata>
+    </ItemGroup>
+    
+    <CheckForDuplicateItemMetadata
+      Items="@(_ResolvedCopyLocalPublishAssetsWithMetadata);@(ResolvedFileToPublish)"
+      MetadataName="RelativePath" >
+      <Output TaskParameter="DuplicateItems" ItemName="_DuplicateResolvedCopyLocalPublishAssets" />
+    </CheckForDuplicateItemMetadata>
+
+    <ItemGroup>
+      <_ResolvedCopyLocalPublishAssets Remove="@(_ResolvedCopyLocalPublishAssets)"/>
+      <_ResolvedCopyLocalPublishAssets Include="@(_ResolvedCopyLocalPublishAssetsWithMetadata)" Exclude="@(_DuplicateResolvedCopyLocalPublishAssets)"/>
```

Comment:
Will this remove *all* copies? In other words, instead of two copies of some file, will we have zero copies of it?

Reply:
This removes all duplicated files from `_ResolvedCopyLocalPublishAssets`, which is fine as those files still exist in `ResolvedFileToPublish`. Later in the logic these item groups will be concatenated together anyway.